lib/request.js

```diff
@@ -417,7 +417,13 @@ module.exports = {
 
   get subdomains() {
     var offset = this.app.subdomainOffset;
-    return (this.host || '')
+    var ip = /^(\d{1,3}\.){3}\d{1,3}$/gim;
```

Comment:
AFAIK, this regex might just be a pragmatic solution, but I'm open to better suggestions.